src/BalancedBagging.jl

```diff
@@ -0,0 +1,232 @@
+        get_majority_minority_inds_counts(y)
+    # get as much balanced subsets as needed
+    X_y_list_s = [
+        get_some_balanced_subset(
+            Xs,
+            ys,
+            majority_inds,
+            minority_inds,
+            majority_count,
+            minority_count,
+            rng,
+        ) for rng in rngs
+    ]
+    # Make a machine for each
+    machines = (machine(:model, Xsub, ysub) for (Xsub, ysub) in X_y_list_s)
+    # Average the predictions from nodes
+    all_preds = [MLJBase.predict(mach, Xs) for (mach, (X, _)) in zip(machines, X_y_list_s)]
+    println(all_preds)
+    yhat = mean(all_preds)
+    return (; predict = yhat)
+end
+### To register with MLJ
+MMI.metadata_pkg(
+    BalancedBaggingClassifier,
+    name = "MLJBalancing",
+    package_uuid = "45f359ea-796d-4f51-95a5-deb1a414c586",
+    package_url = "https://github.com/JuliaAI/MLJBalancing.jl",
+    is_pure_julia = true,
+)
+
+MMI.metadata_model(
+    BalancedBaggingClassifier,
+    input_scitype = Union{Union{Infinite,Finite}},
+    output_scitype = Union{Union{Infinite,Finite}},
+    target_scitype = AbstractVector,
+    load_path = "MLJBalancing." * string(BalancedBaggingClassifier),
+)
+
+MMI.iteration_parameter(::Type{<:BalancedBaggingClassifier{P}}) where {P} =
+    MLJBase.prepend(:model, iteration_parameter(P))
+for trait in [
+    :input_scitype,
+    :output_scitype,
+    :target_scitype,
+    :fit_data_scitype,
+    :predict_scitype,
+    :transform_scitype,
+    :inverse_transform_scitype,
+    :is_pure_julia,
+    :supports_weights,
+    :supports_class_weights,
+    :supports_online,
+    :supports_training_losses,
+    :is_supervised,
+    :prediction_type,
+]
+    quote
+        MMI.$trait(::Type{<:BalancedBaggingClassifier{P}}) where {P} = MMI.$trait(P)
+    end |> eval
+end
+
+"""
+$(MMI.doc_header(BalancedBaggingClassifier))
+
+    Given a probablistic classifier.`BalancedBaggingClassifier` performs bagging by only undersampling
+    majority data in each bag so that its includes as much samples as in the minority data.
+    When the classifier is Adaboost, it is tantamount to EasyEnsemble as presented in Xu-Ying Liu, 
+    Jianxin Wu, & Zhi-Hua Zhou. (2009). Exploratory Undersampling for Class-Imbalance Learning. 
+    IEEE Transactions on Systems, Man, and Cybernetics, Part B (Cybernetics), 39 (2), 539–5501 
+
+
+    # Training data
+
+    In MLJ or MLJBase, bind an instance `model` to data with
+    
+        mach = machine(model, X, y)
+    
+    where
+    
+    - `X`: any table of input features (e.g., a `DataFrame`) so long as elements in each column
+        are subtypes of either the `Finite` or `Infinite` scientific types.
+    
+    - `y`: the binary target, which can be any `AbstractVector` where `length(unique(y)) == 2`
+
+
+    Train the machine with `fit!(mach, rows=...)`.
+
+
+    # Hyperparameters
+
+    - `classifier<:Probabilistic`: The classifier to use to train on each bag.
```

Comment:
I think you called this `model` not `classifier`

```suggestion
    - `model<:Probabilistic`: The classifier to use to train on each bag.
```

Reply:
Indeed.